src/main/kotlin/ChattORE.kt

```diff
@@ -1,107 +1,159 @@
-import commands.*
-import entity.ChattORESpec
-import listener.ChatListener
+import com.velocitypowered.api.event.Subscribe
+import com.velocitypowered.api.event.proxy.ProxyInitializeEvent
+import com.velocitypowered.api.plugin.Dependency
+import com.velocitypowered.api.plugin.Plugin
+import com.velocitypowered.api.plugin.annotation.DataDirectory
+import com.velocitypowered.api.proxy.ProxyServer
+import chattore.commands.*
+import chattore.entity.ChattORESpec
+import chattore.listener.ChatListener
+import chattore.listener.DiscordListener
+import net.kyori.adventure.text.Component
+import net.kyori.adventure.text.serializer.plain.PlainTextComponentSerializer
 import net.luckperms.api.LuckPerms
 import net.luckperms.api.LuckPermsProvider
-import net.md_5.bungee.api.chat.BaseComponent
-import net.md_5.bungee.api.plugin.Plugin
 import org.javacord.api.DiscordApi
 import org.javacord.api.DiscordApiBuilder
+import org.javacord.api.entity.channel.TextChannel
+import org.javacord.api.entity.message.MessageBuilder
+import org.slf4j.Logger
 import java.io.File
+import java.nio.file.Path
 import java.util.*
-import java.util.logging.Level
 
-class ChattORE : Plugin() {
+private const val VERSION = "0.1.0-SNAPSHOT"
+
+@Plugin(
+    id = "chattore",
+    name = "ChattORE",
+    version = VERSION,
+    url = "https://openredstone.org",
+    description = "Because we want to have a chat system that actually wOREks for us.",
+    authors = ["Nickster258", "PaukkuPalikka", "StackDoubleFlow"],
+    dependencies = [Dependency(id = "luckperms")]
+)
+class ChattORE @Inject constructor(val proxy: ProxyServer, val logger: Logger, @DataDirectory dataFolder: Path) {
     lateinit var luckPerms: LuckPerms
     lateinit var config: Config
     private val replyMap: MutableMap<UUID, UUID> = hashMapOf()
     private var discordMap: Map<String, DiscordApi> = hashMapOf()
-    override fun onEnable() {
+    private val dataFolder = dataFolder.toFile()
+
+    @Subscribe
+    fun onProxyInitialization(event: ProxyInitializeEvent) {
         config = loadConfig()
-        BungeeCommandManager(this).apply {
+        VelocityCommandManager(proxy, this).apply {
             registerCommand(Chattore(this@ChattORE))
             registerCommand(HelpOp(this@ChattORE))
-            registerCommand(Me(config, this.plugin.proxy))
-            registerCommand(Message(config, this.plugin.proxy, replyMap))
-            registerCommand(Reply(config, this.plugin.proxy, replyMap))
+            registerCommand(Me(config, this@ChattORE))
+            registerCommand(Message(config, proxy, replyMap))
+            registerCommand(Reply(config, proxy, replyMap))
             setDefaultExceptionHandler(::handleCommandException, false)
         }
         discordMap = loadDiscordTokens()
+        discordMap.forEach { (_, discordApi) -> discordApi.updateActivity(config[ChattORESpec.discord.playingMessage]) }
+        discordMap.values.firstOrNull()?.addListener(DiscordListener(this))
```

Comment:
This fires the event for all channels right? Could attach it to the game-chat channel instead